### PR TITLE
[SPARK-25737][CORE] Remove JavaSparkContextVarargsWorkaround

### PR DESCRIPTION
**Pull request by srowen**
## What changes were proposed in this pull request?

Remove JavaSparkContextVarargsWorkaround

## How was this patch tested?

Existing tests.

### CODE REVIEW

**Comment by kiszk** [comment]
retest this please

**Comment by srowen** [comment]
I think it's a legit failure. It touches a Kinesis test, and that has triggered what I assume is an existing problem in the Kinesis integration. It looks like it didn't like the fact that we updated jackson in `master`, but that change hadn't triggered this test. I'll follow up and see about harmonizing or fixing it first.

**Review by holdenk — COMMENTED** [review]
Python changes look reasonable, one thing we might consider is that there is a helper method `_new_java_array(pylist, java_class)` in `ml/wrapper.py` for the array copying, might make sense to move that up to core. I live in fear off copy'pasta code having an off-by-one mistake but minor.

**Comment by holdenk** [comment]
Jenkins, retest this please.

File: project/MimaExcludes.scala [hunk 1]
```diff
@@ -55,9 +59,12 @@ object MimaExcludes {
     ProblemFilters.exclude[DirectMissingMethodProblem]("org.apache.spark.api.java.JavaSparkContext.accumulable"),
     ProblemFilters.exclude[DirectMissingMethodProblem]("org.apache.spark.api.java.JavaSparkContext.doubleAccumulator"),
     ProblemFilters.exclude[DirectMissingMethodProblem]("org.apache.spark.api.java.JavaSparkContext.accumulator"),
+    // [SPARK-24109] Remove class SnappyOutputStreamWrapper
```
Comment (srowen) [inline project/MimaExcludes.scala]: I retroactively added these from previous changes

**Comment by srowen** [comment]
Merged to master

**Pull request closed without merging**